graph/api/src/main/java/org/jboss/windup/graph/model/resource/FileModel.java

```diff
@@ -222,11 +222,21 @@ OnParseError fromName(String name){
      * need to know which application a particular file is associated with.
      *
      * This is a shortcut for calling getProjectModel().getRootProjectModel().
+     *
+     * Note: In the case of a shared library, this may return the "shared-libs" application.
      */
     @JavaHandler
     ProjectModel getApplication();
 
     /**
+     * Gets the list of all applications that this file is a part of. This will include both the "shared-libs" project
+     * as well as any actual applications.
+     */
+    @JavaHandler
+    @Override
+    Iterable<ProjectModel> getApplications();
+
+    /**
```

Comment:
Again, I find this quite confusing.

```
A
|--- B
|--- C
```
If I call `getApplication()` on A, what should I get? A? And for `getApplications()` result would be also A?
If I call `getApplication()` on B, what should I get? B? And for `getApplications()` result would be A, B?


(more complicated example with shared libs:)
```
A
|--- B
|    |--- C
|    |    |---D
|--- E
|    |--- C
|    |    |---F
```
Now we have 2 different instances of C. Lets call them C1 and C2.

If I call `getApplication()` on C1, result should be B, right? And for `getApplications()` result should be A, B, C1?
If I call `getApplication()` on C2, what should I get? E? And for `getApplications()` result would be A, E, C2?

And for D, what is result of `getApplication()`? `shared-libs` ? And `getApplications()` is A, B, C1, shared-libs?
And for F, what is result of `getApplication()`? `shared-libs` ? And `getApplications()` is A, E, C2, shared-libs?

Reply:
I'm just adding a few notes from the discussion from this morning for reference.

In the first case, it would always return A. In the second case, any call for "C" would return both "A" and the synthetic "shared-libs" application.

Calls for "D" would do exactly the same thing (return "A" and the synthetic shared-libs application).